doc/schemas/cchost_config.json

```diff
@@ -0,0 +1,422 @@
+          "node_to_node_address_file": {
+            "type": "string",
+            "description": "Path to file in which node address (hostname and port) will be written to on startup. This option is particularly useful when binding to port 0 and getting auto-assigned a port by the OS. No file is created if this entry is not specified"
+          },
+          "rpc_addresses_file": {
+            "type": "string",
+            "description": "Path to file in which all RPC addresses (hostnames and ports) will be written to on startup. This option is particularly useful when binding to port 0 and getting auto-assigned a port by the OS. No file is created if this entry is not specified"
+          }
+        },
+        "description": "This section includes configuration for additional files output by the node"
+      },
+      "tick_interval": {
+        "type": "string",
+        "default": "10ms",
+        "description": "Interval at which the enclave time will be updated by the host (modify with care!)"
+      },
+      "slow_io_logging_threshold": {
+        "type": "string",
+        "default": "10000us",
+        "description": "Maximum duration of I/O operations (ledger and snapshots) after which slow operations will be logged to node log"
+      },
+      "node_client_interface": {
+        "type": "string",
+        "description": "Address to bind to for node-to-node client connections. If unspecified, this is automatically assigned by the OS. This option is particularly useful for testing purposes (e.g. establishing network partitions between nodes)"
+      },
+      "client_connection_timeout": {
+        "type": "string",
+        "default": "2000ms",
+        "description": "Maximum duration after which unestablished client connections will be marked as timed out and either re-established or discarded"
+      },
+      "worker_threads": {
+        "type": "integer",
+        "default": "0",
+        "description": "Experimental. Number of additional threads processing incoming client requests in the enclave (modify with care!)"
+      },
+      "memory": {
+        "type": "object",
+        "properties": {
+          "circuit_size": {
+            "type": "string",
+            "default": "4MB",
+            "description": "Size (size string) of the internal host-enclave ringbuffers, as a power of 2"
+          },
+          "max_msg_size": {
+            "type": "string",
+            "default": "16MB",
+            "description": "Maximum size (size string) for a message sent over the ringbuffer, as a power of 2. Messages may be split into multiple fragments, but this limits the total size of the sum of those fragments"
+          },
+          "max_fragment_size": {
+            "type": "string",
+            "default": "64KB",
+            "description": "Maximum size (size string) of individual ringbuffer message fragments, as a power of 2. Messages larger than this will be split into multiple fragments"
+          }
```

Comment:
It's odd that these descriptions say "as a power of 2". I think that's an artifact from when we were passing 8 and turning that into 2^8. Perhaps should say "must be a power of 2" (if that's still a requirement)?

Reply:
Good point. I've re-phrased this and removed the note where the requirement is not applicable (i.e. max msg/fragment size).